docs/invoke.mdx

```diff
@@ -22,71 +18,587 @@ You can visualize your state machines and easily invoke actors in our drag-and-d
+- `systemId` - A string identifing the actor, unique system-wide.
+
+### Source {#src}
+
+The `src` represents the [actor logic](actors.mdx#actor-logic-creators) the machine should use when creating the actor. There are several actor logic creators available in XState:
+
+- [State machine logic (`createMachine`)](actors.mdx#createmachine)
+- [Promise logic (`fromPromise`)](actors.mdx#frompromise), where invoke will take the `onDone` transition on `resolve`, or the `onError` transition on `reject`
+- [Transition function logic (`fromTransition`)](actors.mdx#fromtransition), which follows the reducer pattern
+- [Observable logic (`fromObservable`)](actors.mdx#fromobservable), which can send events to the parent machine, and where invoke will take an `onDone` transition when completed
+- [Event observable logic (`fromEventObservable`)](actors.mdx#fromeventobservable), like Observable logic but for streams of event objects
+- [Callback logic (`fromCallback`)](actors.mdx#fromcallback), which can send events to and receive events from the parent machine
+
+The invoke `src` can be _inline_ or _provided_.
+
+#### Inline `src`
+
+Either directly inline:
 
-## Actor snapshots
+```ts
+invoke: {
+  src: fromPromise(…)
+}
+```
 
-The actor snapshot is the latest emitted value from the actor. It can be read from `actorRef.getSnapshot()`.
+Or from some logic in the same scope as the machine:
 
-- The snapshot is not the same as the internal state; it is what the actor chooses to share with observers.
+```ts
+const logic = fromPromise(…)
+const machine = createMachine({
+  // …
+  invoke: {
+    src: logic
+  }
+});
+```
 
-## `onDone`
+#### Provided `src`
+
+The `src` can be [provided in the machine implementation](machines.mdx#providing-implementations) and referenced using a string or an object.
+
+```ts
+const machine = createMachine({
+  // …
+  invoke: {
+    src: 'workflow' // string reference
+  }
+});
+
+const actor = createActor(machine.provide({
+  actors: {
+    workflow: fromPromise(…) // provided
+  }
+}));
+```
+
+### `onDone`
 
 - Transitions when invoked actor is complete
-- Event gets `.output` with actor's output data
+- Event object `output` property is provided with actor's output data
+- Not available for callback actors
+
+:::warning
+Don't get the `onDone` property on a state confused with `invoke.onDone` - they are similar transitions, but refer to different things.
+
+- The `onDone` property on a [state](states.mdx) node refers to the compound state node reaching a [final state](final-states.mdx).
+- The `invoke.onDone` property refers to the invocation (`invoke.src`) being done.
+
+```js {5,13}
+// ...
+loading: {
+  invoke: {
+    src: someSrc,
+    onDone: {/* ... */} // refers to `someSrc` being done
+  },
+  initial: 'loadFoo',
+  states: {
+    loadFoo: {/* ... */},
+    loadBar: {/* ... */},
+    loadingComplete: { type: 'final' }
+  },
+  onDone: 'loaded' // refers to 'loading.loadingComplete' being reached
+}
```

Comment:
```suggestion
```js
// ...
loading: {
  invoke: {
    src: someSrc,
    // refers to `someSrc` being done
    // highlight-next-line
    onDone: {/* ... */}
  },
  initial: 'loadFoo',
  states: {
    loadFoo: {/* ... */},
    loadBar: {/* ... */},
    loadingComplete: { type: 'final' }
  },
  // refers to 'loading.loadingComplete' being reached
  // highlight-next-line
  onDone: { target: 'loaded' }
}
```